File: csharp/TicketOfficeService.Tests/Hooks/TestHooks.cs
using Reqnroll;
using TicketOfficeService.Tests.Support;

namespace TicketOfficeService.Tests.Hooks;

[Binding]
public class TestHooks
{
    private readonly TestContext _context;

    public TestHooks(TestContext context)
    {
        _context = context;
    }

    [BeforeScenario]
    public void BeforeScenario()
    {
        _context.Reset();
    }

    [AfterScenario]
    public void AfterScenario()
    {
        _context.Reset();
    }
}

File: java-springboot/src/test/java/com/trainreservation/hooks/TestHooks.java
package com.trainreservation.hooks;

import com.trainreservation.support.TestContext;
import io.cucumber.java.Before;
import org.springframework.beans.factory.annotation.Autowired;

public class TestHooks {

    @Autowired
    private TestContext context;

    @Before
    public void beforeScenario() {
        context.reset();
    }
}
